### PR TITLE
[chore] Release v0.73.0

### LINKED ISSUE #1562: Release the operator v0.73.0

### PR DESCRIPTION
**Pull request by jaronoff97**
Resolves #1562

### CODE REVIEW

File: bundle.Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=opentelemetry-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.27.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.28.0
```
Comment (frzifus) [inline bundle.Dockerfile:9]: I assume you did not intend to update operator-sdk? :sweat_smile:

File: autoinstrumentation/python/requirements.txt [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
-opentelemetry-instrumentation-sklearn==0.36b0
-opentelemetry-instrumentation-sqlalchemy==0.36b0
-opentelemetry-instrumentation-sqlite3==0.36b0
-opentelemetry-instrumentation-starlette==0.36b0
-opentelemetry-instrumentation-tornado==0.36b0
-opentelemetry-instrumentation-tortoiseorm==0.36b0
-opentelemetry-instrumentation-urllib==0.36b0
-opentelemetry-instrumentation-urllib3==0.36b0
-opentelemetry-instrumentation-wsgi==0.36b0
+opentelemetry-instrumentation-aio-pika==0.37b0
```
Comment (frzifus) [inline autoinstrumentation/python/requirements.txt:15]: do we need to update this too?

Comment (jaronoff97) [reply autoinstrumentation/python/requirements.txt:15]: part of the instructions were to bump all the auto-inst to latest.

File: bundle.Dockerfile (hunk 1 above)
Comment (jaronoff97) [reply bundle.Dockerfile:9]: my local version upgraded, i can force to use 1.27 though if we are worried about the upgrades.

File: autoinstrumentation/nodejs/package.json
Comment (pavolloffay) [inline autoinstrumentation/nodejs/package.json:1]: Is there any reason why the nodejs agent is updated in this PR? 

I expect that there shouldn't be changes in this file.

File: autoinstrumentation/python/requirements.txt (hunk 2 above)
Comment (pavolloffay) [reply autoinstrumentation/python/requirements.txt:15]: No, as part of the release we should bump only the released auto-instrumentation agents in versions.txt

File: bundle.Dockerfile (hunk 1 above)
Comment (pavolloffay) [reply bundle.Dockerfile:9]: please extract this to a separete PR

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -31,13 +31,13 @@ metadata:
     categories: Logging & Tracing
     certified: "false"
     containerImage: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator
-    createdAt: "2023-03-14T11:39:39Z"
+    createdAt: "2023-03-16T02:42:25Z"
     description: Provides the OpenTelemetry components, including the Collector
-    operators.operatorframework.io/builder: operator-sdk-v1.27.0
+    operators.operatorframework.io/builder: operator-sdk-v1.28.0
```
Comment (pavolloffay) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:36]: please extract this to a separete PR

File: bundle/metadata/annotations.yaml [hunk 4]
```diff
@@ -5,7 +5,7 @@ annotations:
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: opentelemetry-operator
   operators.operatorframework.io.bundle.channels.v1: alpha
-  operators.operatorframework.io.metrics.builder: operator-sdk-v1.27.0
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.28.0
```
Comment (pavolloffay) [inline bundle/metadata/annotations.yaml:8]: please extract this to a separete PR

File: config/manager/kustomization.yaml [hunk 5]
```diff
@@ -1,2 +1,8 @@
 resources:
 - manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (pavolloffay) [inline config/manager/kustomization.yaml:3]: no change here

**Review by pavolloffay — COMMENTED** [review]
The release managers table in release.md needs to be updated.

**Comment by TylerHelmuth** [comment]
I would absolutely love to get this PR merged before 0.73.0 is release: https://github.com/open-telemetry/opentelemetry-operator/pull/1582.  This PR unblocks an issue we're having in the helm charts with the e2e tests.

**Pull request merged**